server/datastore/mysql/hosts.go

```diff
@@ -748,3 +777,24 @@ func (d *Datastore) SaveHostAdditional(host *fleet.Host) error {
 
 	return nil
 }
+
+func (d *Datastore) SaveHostUsers(host *fleet.Host) error {
+	if len(host.Users) == 0 {
+		return nil
+	}
+
+	var args []interface{}
+	for _, u := range host.Users {
+		args = append(args, host.ID, u.Uid, u.Username, u.Type, u.GroupName)
+	}
+	values := strings.TrimSuffix(strings.Repeat("(?, ?, ?, ?, ?),", len(host.Users)), ",")
+	sql := fmt.Sprintf(
+		`INSERT IGNORE INTO host_users (host_id, uid, username, user_type, groupname) VALUES %s`,
+		values,
+	)
+	if _, err := d.db.Exec(sql, args...); err != nil {
+		return errors.Wrap(err, "insert users")
+	}
```

Comment:
Do we need a `DELETE` somewhere in here? Or should we perhaps consider using the first detected/deleted timestamp strategy discussed during our vulnerability management conversation yesterday?

Reply:
Hm, that's a good point. Yeah, let's add the timestamp approach.